llvm/lib/CodeGen/MachineLICM.cpp

```diff
@@ -352,6 +386,11 @@ bool MachineLICMBase::runOnMachineFunction(MachineFunction &MF) {
   MRI = &MF.getRegInfo();
   SchedModel.init(&ST);
 
+  // FIXME: Remove this assignment or convert to an assert?
+  // MachineLICM and PostRAMachineLICM were distinguished by introducing
+  // EarlyMachineLICM and MachineLICM respectively to avoid "using an unreliable
+  // MRI::isSSA() check to determine whether register allocation has happened"
+  // (See 4a7c8e7).
   PreRegAlloc = MRI->isSSA();
```

Comment:
What about `MachineFunctionProperties::Property::IsSSA` or `MachineFunctionProperties::Property::NoVRegs`?

Reply:
IIRC MRI::isSSA is implemented with the property check

Reply:
If we want to check if reg alloc has happened with the property check, there is no point in keeping the `PreRegAlloc` flag as a constructor parameter.
Why was isSSA() called an unreliable check? @MatzeB

Reply:
It's not an unreliable check